From cb478b93dc44e4ef805ca92fc6e2133fa30153a4 Mon Sep 17 00:00:00 2001
From: Zhen Lei <thunder.leizhen@huawei.com>
Date: Tue, 8 Mar 2022 22:06:00 +0800
Subject: [PATCH] arm64: replace is_compat_task() with is_ilp32_compat_task()
 in TASK_SIZE_MAX
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cb478b93dc44e4ef805ca92fc6e2133fa30153a4
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4V6F8
CVE: NA

-------------------------------------------------------------------------

access_ok() is used to preliminarily check 'uaddr' to avoid unnecessary
page fault caused by invalid input. The page fault will do the accurate
address verification based on task's mm. It's also used to do a check on
the get_fs(), see the comments of __access_ok().

But now the support for get_fs() on arm64 has been deleted by commit
edf84200127a ("arm64: uaccess: remove set_fs()"). So access_ok()
does not need to perform such strict checks for compat tasks.

Remove the is_compact_task() check can improve the performance of
syscalls. For example, all test items of libMicro can be improved by
4.89% on average.

The next patch will avoid calling is_ilp32_compat_task() by default by
close its build option, because ILP32 has specific requirements.

Signed-off-by: Zhen Lei <thunder.leizhen@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Reviewed-by: liuchao (CR) <liuchao173@huawei.com>
Reviewed-by: Liu Chao (CR) < liuchao173@huawei.com >
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/include/asm/processor.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/include/asm/processor.h b/arch/arm64/include/asm/processor.h
index 8e6f1af816c9..617cdd40c61b 100644
--- a/arch/arm64/include/asm/processor.h
+++ b/arch/arm64/include/asm/processor.h
@@ -49,7 +49,7 @@
 #define TASK_SIZE_64		(UL(1) << vabits_actual)
 
 #ifdef CONFIG_COMPAT
-#define TASK_SIZE_MAX		(is_compat_task() ? \
+#define TASK_SIZE_MAX		(is_ilp32_compat_task() ? \
 				UL(0x100000000) : (UL(1) << VA_BITS))
 #if defined(CONFIG_ARM64_64K_PAGES) && defined(CONFIG_KUSER_HELPERS)
 /*
-- 
2.26.2

